### DISCUSSION TITLE
Refreshing OAuth token for http sink

### DISCUSSION THREAD

**Discussion by mans2singh** [vectordotdev/vector#17192] [Q&A] [answered]
Hey Folks:

I am using vector [http sink](https://vector.dev/docs/reference/configuration/sinks/http/#auth.strategy) and it has strategy to use OAuth token for authentication.  However I could not find any documentation on how the token can be refreshed when it expires.  I believe some sinks do support that but can http sink be configured to refresh it ?  

If you have any other advice, please let me know.

Thanks

**Comment by bruceg** [comment, marked as answer]
Sorry, no, the http sink does not have support for refreshing the OAuth token.

**Reply by mans2singh** [reply]
Thanks @bruceg

**Comment by earonesty** [comment]
this works for http-client: https://github.com/logtura/logtura-http-client
